frontend/public/components/chargeback.tsx

```diff
@@ -230,68 +229,55 @@ class ReportData extends React.Component<ReportDataProps, ReportDataState> {
   }
 
   componentWillReceiveProps(nextProps) {
-    if (this.state.inFlight) {
+    // if request is still inFlight or the report has finished running, then return
+    if (this.state.inFlight || this.state.isReportFinished) {
       return;
     }
     const conditions = _.get(nextProps.obj, 'status.conditions');
-    const isFinished = _.some(conditions, {type: 'Running', status: 'False'});
-    if (isFinished) {
-      this.fetchData();
+    const isReportFinished = _.some(conditions, {type: 'Running', status: 'False'});
+    if (isReportFinished) {
+      this.setState({
+        isReportFinished: true,
+      }, () => this.fetchData());
     }
   }
 
   makeTable(data=this.state.data) {
-    const keys = this.filterKeys(data);
-    const sortBy = getQueryArgument('sortBy') || keys[1] || keys[0];
-    const orderBy = getQueryArgument('orderBy') || (sortBy === keys[0] ? 'asc' : 'desc');
-    const rows = this.transformData(data, sortBy, orderBy);
+    if (_.isEmpty(data)) {
+      return;
+    }
+    const cols = this.getColumns(data);
+    const rows = this.getRows(data);
 
     this.setState({
       data,
-      sortBy,
-      orderBy,
-      keys,
+      cols,
       rows,
     });
   }
 
-  orderBy(col) {
-    setQueryArgument('orderBy', col);
-    this.makeTable();
-  }
-
-  sortBy(col) {
-    setQueryArgument('sortBy', col);
-    this.makeTable();
-  }
-
-  filterKeys(data=[]) {
-    const keys = _.keys(data[0]).filter(k => {
+  getColumns(data=[]) {
+    const cols = _.keys(data[0]).filter(k => {
       if (COLS_BLACK_LIST.has(k)) {
```

Comment:
Not new but better to return `!` the if expression directly

Reply:
?? sorry, not following.  Can u pls elaborate?  -thanks

Reply:
Ok to ignore this for now?  Follow up PR (already in progress) removes the COLS_BLACK_LIST check.